### ISSUE TITLE
Thunderbolt as an Entity

### ISSUE THREAD

**Issue by Tri125** [mc-server/MCServer#1982]
I was looking over CastThunderbolt in World.cpp and wondered if it would be best to create an entity class for Thunderbolt altogether.

It can damage, it can create fire on impact and detonate TNT, it Broadcast two different type of sounds, etc.

Thoughts?

**Comment by worktycho** [comment]
Its probably not a good idea to create an entity class as the cEntity class is heavily tied into the protocol definition of an Entity, which a thunderbolt isn't in. The piston approach might match this best, where it has a separate class.

**Comment by pwnorbitals** [comment]
This is a very old issue. A decision should be taken or this issue closed. imo separating the thunderbolt in its own class would be the best wby to go

**Comment by tigerw** [comment]
Yes, we will be doing this. The (namespace probably) will be placed in Physics\

**Comment by Pokechu22** [comment]
As of 1.16, thunderbolts are now normal entities to the Notchian client (and use Spawn Entity now if I recall correctly), instead of the weird weather pseudo-entity thing they were before.

**Comment by tigerw** [comment]
Thanks, do you know if we need to explicitly destroy the entity or does the client handle that?

**Comment by Pokechu22** [comment]
Nothing appears to have changed in that regard for 1.16; as far as I can tell the client removes it itself (but the server might also send a remove packet?)
